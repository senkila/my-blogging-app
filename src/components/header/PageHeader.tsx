import Image from 'next/image'
import Link from 'next/link'
import './header.css'

export const PageHeader = ({ className }: { className?: string }) => {
    return (
        <div>
            <Link href={'/'}>
                <Image
                    src="/header-text.svg"
                    alt="Embracing the Imposter"
                    width={1000}
                    height={150}
                    priority
                    className="heading"
                />
                <Image
                    src="/subheader.svg"
                    alt="A journey from no ops to dev op and beyond"
                    width={1000}
                    height={80}
                    priority
                    className="flavourText"
                />
            </Link>
        </div>
    )
}

export default PageHeader
