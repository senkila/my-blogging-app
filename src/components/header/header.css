@media (max-width: 600px) {
    .flavourText {
        display: none;
    }
}
