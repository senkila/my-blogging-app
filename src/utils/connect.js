const { MongoClient } = require('mongodb')

// Replace the following with your Atlas connection string
export const mongoDbUrl =
    'mongodb+srv://csengly:hks11c4Fgt9T1lV3@senki-test-01.glsgxqb.mongodb.net/?retryWrites=true&w=majority&appName=senki-test-01'

// Connect to your Atlas cluster
const client = new MongoClient(url)

async function run() {
    try {
        await client.connect()
        console.log('Successfully connected to Atlas')
    } catch (err) {
        console.log(err.stack)
    } finally {
        await client.close()
    }
}

run().catch(console.dir)
