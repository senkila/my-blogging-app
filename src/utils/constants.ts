const hostUrl = 'http://localhost:3000/'

export { hostUrl }
