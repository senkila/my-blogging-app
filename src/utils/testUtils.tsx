import React, { ReactElement } from 'react'
import { render, RenderOptions } from '@testing-library/react'
import { AppContextProvider } from '../components'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
    return <AppContextProvider>{children}</AppContextProvider>
}

const customRender = (
    ui: ReactElement,
    options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
