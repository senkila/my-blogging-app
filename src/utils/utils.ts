import short from 'short-uuid'

const getCurrentDate = () => {
    const date = new Date()
    return date.toISOString()
}

const covertToLocalDate = (dateString: string) => {
    const date = new Date(dateString)
    return date.toLocaleDateString('en-AU')
}

const generateId = (title: string) => {
    const shortuuid = short.generate()
    const pageId = `${title.toLocaleLowerCase().replaceAll(' ', '-')}-${shortuuid}`
    return pageId
}

const isHighlighted = (el: HTMLElement) => {
    return el.getAttribute('data-highlighted') === 'yes'
}

export { getCurrentDate, covertToLocalDate, generateId, isHighlighted }
