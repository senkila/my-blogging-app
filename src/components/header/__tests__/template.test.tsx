import '@testing-library/jest-dom'
import React from 'react'
import { render, screen } from '@/src/utils/testUtils'
import { PageHeader } from '../header'

describe('PageHeader', () => {
    it('display content as html', () => {
        const ui = <PageHeader />
        render(ui)
        const image = screen.getAllByRole('img')
        expect(image).toHaveLength(2)
        expect(image[0]).toMatchSnapshot()
        expect(image[1]).toMatchSnapshot()
    })
})
