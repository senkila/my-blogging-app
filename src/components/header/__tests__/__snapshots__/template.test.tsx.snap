// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader display content as html 1`] = `
<img
  alt="Embracing the Imposter"
  class="heading"
  data-nimg="1"
  decoding="async"
  fetchpriority="high"
  height="150"
  src="/header-text.svg"
  style="color: transparent;"
  width="1000"
/>
`;

exports[`PageHeader display content as html 2`] = `
<img
  alt="A journey from no ops to dev op and beyond"
  class="flavourText"
  data-nimg="1"
  decoding="async"
  fetchpriority="high"
  height="80"
  src="/subheader.svg"
  style="color: transparent;"
  width="1000"
/>
`;
